tests/e2e/mapper/mapper_suite_test.go

```diff
@@ -0,0 +1,246 @@
+			req.Header.Set("Content-Type", "application/yaml")
+			resp, err := client.Do(req)
+			Expect(err).To(BeNil())
+			Expect(resp.StatusCode).Should(Equal(http.StatusCreated))
+			utils.InfoV6("%s %s %v in %v", req.Method, req.URL, resp.Status, time.Now().Sub(t))
+		}
+
+		//updating deployment file with edgenode name and dockerhubusername
+		cmd = exec.Command("bash", "-x", "scripts/bluetoothconfig.sh", ctx.Cfg.DockerHubUserName, nodeName)
+		err = utils.PrintCombinedOutput(cmd)
+		Expect(err).Should(BeNil())
+	})
+
+	AfterSuite(func() {
+		By("After Suite Execution....!")
+
+		//Delete created deployment
+		var podlist metav1.PodList
+		var deploymentList v1.DeploymentList
+		var UID string = "bluetooth-device-mapper-deployment"
+		err := utils.GetDeployments(&deploymentList, ctx.Cfg.K8SMasterForKubeEdge+DeploymentHandler)
+		Expect(err).To(BeNil())
+		for _, deployment := range deploymentList.Items {
+			if deployment.Name == UID {
+				label := nodeName
+				podlist, err = utils.GetPods(ctx.Cfg.K8SMasterForKubeEdge+AppHandler, label)
+				Expect(err).To(BeNil())
+				StatusCode := utils.DeleteDeployment(ctx.Cfg.K8SMasterForKubeEdge+DeploymentHandler, deployment.Name)
+				Expect(StatusCode).Should(Equal(http.StatusOK))
+			}
+		}
+		utils.CheckPodDeleteState(ctx.Cfg.K8SMasterForKubeEdge+AppHandler, podlist)
+
+		//Delete config map created
+		go utils.HandleConfigmap(chconfigmapRet, http.MethodDelete, ctx.Cfg.K8SMasterForKubeEdge+ConfigmapHandler+"/"+"device-profile-config-edge-node1", false)
+
+		//Deregister the edge node from master
+		err = utils.DeRegisterNodeFromMaster(ctx.Cfg.K8SMasterForKubeEdge+NodeHandler, nodeName)
+		Expect(err).Should(BeNil())
+		Eventually(func() int {
+			statuscode := utils.CheckNodeDeleteStatus(ctx.Cfg.K8SMasterForKubeEdge+NodeHandler, nodeName)
+			utils.Info("Node Name: %v, Node Statuscode: %v", nodeName, statuscode)
+			return statuscode
+		}, "60s", "4s").Should(Equal(http.StatusNotFound), "Node register to the k8s master is unsuccessfull !!")
+
+		//Deleting the created devicemodel and deviceinstance
+		client := &http.Client{}
+		req, err := http.NewRequest(http.MethodDelete, ctx.Cfg.K8SMasterForKubeEdge+CrdHandler+"/"+deviceModelCRD, nil)
+		Expect(err).Should(BeNil())
+		req.Header.Set("Content-Type", "application/yaml")
+		resp, err := client.Do(req)
+		Expect(err).Should(BeNil())
+		Expect(resp.StatusCode).Should(Equal(http.StatusOK))
+		req, err = http.NewRequest(http.MethodDelete, ctx.Cfg.K8SMasterForKubeEdge+CrdHandler+"/"+deviceCRD, nil)
+		Expect(err).Should(BeNil())
+		req.Header.Set("Content-Type", "application/yaml")
+		resp, err = client.Do(req)
+		Expect(err).Should(BeNil())
+		Expect(resp.StatusCode).Should(Equal(http.StatusOK))
+
+		//Run the Cleanup steps to kill edge_core and edgecontroller binaries
+		cmd := exec.Command("bash", "-x", "scripts/cleanup.sh")
+		err = utils.PrintCombinedOutput(cmd)
+		Expect(err).Should(BeNil())
+		time.Sleep(2 * time.Second)
```

Comment:
Why do we need sleep at the end of the tests ?

Reply:
Updated